### PR TITLE
Describe the optional dependency Matplotlib

### PR DESCRIPTION
**Pull request by ns-rse**
## Description

Adds explanation of what the optional dependency on Matplotlib offers to the install instructions.

See #7281 

## Checklist

- [x] A descriptive but concise pull request title
- [ ] [~~Docstrings for all functions~~](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] [~~Unit tests~~](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- [ ] ~~A gallery example in `./doc/examples` for new features~~
- [x] [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

```release-note
Adds explanation of what the optional dependency on Matplotlib offers to the install instructions.
```

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -399,6 +399,8 @@ functionality is only available with the following installed:
 * `Dask <https://dask.org/>`__
     The ``dask`` module is used to speed up certain functions.
 
+* `Matplotlib <https://matplotlib.org`__
```
Comment (mkcor) [inline INSTALL.rst:402]:
```suggestion
* `Matplotlib <https://matplotlib.org>`__
```
(reST syntax)

**Review by mkcor — COMMENTED** [review]
Nice addition! I agree that Matplotlib should be mentioned alongside other optional dependencies. Thank you, @ns-rse!

File: INSTALL.rst [hunk 2]
```diff
@@ -399,6 +399,8 @@ functionality is only available with the following installed:
...
+    Used in a number of the ``draw`` functions for generating shapes.
```
Comment (jarrodmillman) [inline INSTALL.rst:403]: Matplotlib is used in more modules than ``draw`` and is used for drawing in general (not just for generating shapes).

File: INSTALL.rst [hunk 3]
```diff
@@ -399,6 +399,9 @@ functionality is only available with the following installed:
 * `Dask <https://dask.org/>`__
     The ``dask`` module is used to speed up certain functions.
 
+* `Matplotlib <https://matplotlib.org>`__
+    Used in a number of the ``draw`` functions, for thresholding, segmentation
+    and as a plugin for reading images.
```
Comment (lagru) [inline INSTALL.rst:404]: Hmm, thanks for incorporating the feedback. Could you maybe rephrase further so that it doesn't sound like its necessary for all thresholding or segmentation functions?

Maybe something like 

```suggestion
    Used in a few functions to add interactivity, some segmentation functions
    as well as a plugin for reading images.
```

should stay up to date longer.. :crossed_fingers: 

As a followup idea, maybe we could modify the [`require` decorator](https://github.com/scikit-image/scikit-image/blob/c459c2adfc2b15e616c6858e034de178f4ba449a/skimage/_shared/version_requirements.py#L111) such that it adds a note to docstrings.

Comment (mkcor) [reply INSTALL.rst:404]: I would keep it even more general, along these lines:

"Used in various functions, e.g., for drawing, segmenting, reading images."

> As a followup idea, maybe we could modify the [require decorator](https://github.com/scikit-image/scikit-image/blob/c459c2adfc2b15e616c6858e034de178f4ba449a/skimage/_shared/version_requirements.py#L111) such that it adds a note to docstrings.

Good idea!

File: INSTALL.rst [hunk 4]
```diff
@@ -399,6 +399,8 @@ functionality is only available with the following installed:
 * `Dask <https://dask.org/>`__
     The ``dask`` module is used to speed up certain functions.
 
+* `Matplotlib <https://matplotlib.org>`__
+    Used in various functions, e.g. for drawing, segmenting, reading images.
```
Comment (mkcor) [inline INSTALL.rst:403]:
```suggestion
    Used in various functions, e.g., for drawing, segmenting, reading images.
```
(by convention)

**Review by mkcor — APPROVED** [review]

File: INSTALL.rst (hunk 3 above)
Comment (mkcor) [reply INSTALL.rst]: > As a followup idea, maybe we could modify the [require decorator](https://github.com/scikit-image/scikit-image/blob/c459c2adfc2b15e616c6858e034de178f4ba449a/skimage/_shared/version_requirements.py#L111) such that it adds a note to docstrings.

@lagru I wanted to create an issue with this content just so we don't forget. But, as I was trying to write the issue title/description, I realized I didn't know how to envision the change: Do we append a note to `name.__doc__`... on the fly? How does it get saved/printed? :thinking:

Comment (lagru) [reply INSTALL.rst]: Yes, modifying the `__doc__` is probably the best idea. Perhaps similar to [`_docstring_add_deprecated`](https://github.com/scikit-image/scikit-image/blob/eb35f51ac173f58dd1deee8c9ecc33efaa09af77/skimage/_shared/utils.py#L332).

**Comment by lagru** [comment]
Thanks @ns-rse!
